### PR TITLE
Rejiggers sticky header.

### LINKED ISSUE #3310: Change or remove sticky header
The current sticky header in the Editor is causing problems in which things are stuck below it. For example:

image

Please try to change the sticky header so that it works like a Github ticket (only becomes sticky when scrolled to top) or remove the sticky feature.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3310

## Why was this change made?
So the sticky header functions as hoped / intended.

## How was this change tested?
Local

### CODE REVIEW

File: src/utilities/Utilities.js [hunk 1]
```diff
@@ -156,3 +156,26 @@ export const isInViewport = (elem) => {
...
+  if (!elem) return false
+  const bounding = elem.getBoundingClientRect()
+  return (
+    bounding.top >= 0 &&
+    bounding.top <=
+      (window.innerHeight || document.documentElement.clientHeight)
+  )
+}
+
+export const stickyScrollIntoView = (selector) => {
```
Comment (jcoyne) [inline src/utilities/Utilities.js:171]: Can you parameterize the selector for the stickyHeader too?  maybe `(targetSelector, headerSelector = "#sticky-resource-header")`

**Comment by peetucket** [comment]
Seems more flexible than my approach in allowing for variable width headers in the smooth scroll.  Do we still want to ensure the z-index of the `.dropdown-menu` class still wins? (https://github.com/LD4P/sinopia_editor/pull/3311/files#diff-d6c5465dec7690d11d69e5841d4517dcea9eddae3932c3be344267fad22a017dR28-R31)

Also, looks like a number of test failures.

**Comment by jcoyne** [comment]
FWIW, I think that setting the zindex for the dropdown menus is way easier to comprehend than the code that determines if the main header is visible.

File: src/utilities/Utilities.js (hunk 1 above)
Comment (peetucket) [reply src/utilities/Utilities.js]: Done.

**Comment by peetucket** [comment]
Fixed the tests by ensuring the selectors that look for the headers are unique (now that the sticky header adds a second copy).

**Review by peetucket — APPROVED** [review]

**Pull request merged**